modules/crm_core_default_matching_engine/src/Plugin/crm_core_match/engine/DefaultMatchingEngine.php

```diff
@@ -78,36 +79,238 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function match(ContactInterface $contact) {
     $ids = array();
-    /* @var \Drupal\crm_core_default_matching_engine\Entity\MatchingRule $matching_rule */
-    $matching_rule = $this->entityManager->getStorage('crm_core_default_engine_rule')->load($contact->bundle());
-    // Check if match is enabled for this contact type.
-    if ($matching_rule->status()) {
-      $fields = $contact->getFieldDefinitions();
-
-      $results = array();
-      foreach ($matching_rule->rules as $name => $rules) {
-        if (isset($fields[$name])) {
-          $rules['field'] = $fields[$name];
-
-          if (!$this->pluginManager->hasDefinition($rules['field']->getType())) {
-            continue;
-          }
-
-          /* @var \Drupal\crm_core_default_matching_engine\Plugin\crm_core_match\field\FieldHandlerInterface $field_handler */
-          $field_handler = $this->pluginManager->createInstance($rules['field']->getType(), $rules);
-
-          foreach ($field_handler->getPropertyNames() as $name) {
-            $results += $field_handler->match($contact, $name);
-          }
+    /* @var \Drupal\crm_core_match\Entity\Matcher $matcher */
+    $matcher = $this->configuration['plugin_config'];
```

Comment:
I'd just pass in the plugin configuration/settings array directly as $configuration. There was an issue to allow objects as $configuration but that never made it in and this is a bit a strange trick. If we want to keep this, then I'd suggest to assign to $this->matcher in the base class __construct()